lib/cli/src/commands/init.rs

```diff
@@ -0,0 +1,472 @@
+        let homepage = cargo_toml.as_ref().and_then(|t| t.homepage.clone());
+        let description = cargo_toml
+            .as_ref()
+            .and_then(|t| t.description.clone())
+            .unwrap_or_else(|| format!("Description for package {module_name}"));
+
+        let default_abi = wapm_toml::Abi::Wasi;
+        let bindings = Self::get_bindings(&target_file, bin_or_lib);
+        let modules = vec![wapm_toml::Module {
+            name: module_name.to_string(),
+            source: cargo_toml
+                .as_ref()
+                .map(|p| {
+                    // Normalize the path to /target/release to be relative to the parent of the Cargo.toml
+                    let outpath = p
+                        .build_dir
+                        .join("release")
+                        .join(&format!("{module_name}.wasm"));
+                    let canonicalized_outpath = outpath.canonicalize().unwrap_or(outpath);
+                    let outpath_str = format!("{}", canonicalized_outpath.display());
+                    let manifest_canonicalized = manifest_path
+                        .parent()
+                        .and_then(|p| p.canonicalize().ok())
+                        .unwrap_or_else(|| manifest_path.clone());
+                    let manifest_str = format!("{}/", manifest_canonicalized.display());
+                    let relative_str = outpath_str.replacen(&manifest_str, "", 1);
+                    Path::new(&relative_str).to_path_buf()
+                })
+                .unwrap_or_else(|| Path::new(&format!("{module_name}.wasm")).to_path_buf()),
+            kind: None,
+            abi: default_abi,
+            bindings: bindings.clone(),
+            interfaces: Some({
+                let mut map = HashMap::new();
+                map.insert("wasi".to_string(), "0.1.0-unstable".to_string());
+                map
+            }),
+        }];
+
+        let constructed_manifest = wapm_toml::Manifest {
+            package: wapm_toml::Package {
+                name: format!("{namespace}/{module_name}"),
+                version,
+                description,
+                license,
+                license_file,
+                readme,
+                repository,
+                homepage,
+                wasmer_extra_flags: None,
+                disable_command_rename: false,
+                rename_commands_to_raw_command_name: false,
+            },
+            dependencies: self.get_dependencies(),
+            command: Self::get_command(&modules, bin_or_lib),
+            module: match bin_or_lib {
+                BinOrLib::Empty => None,
+                _ => Some(modules),
+            },
+            fs: self.get_filesystem_mapping(),
+            base_directory_path: target_file
+                .parent()
+                .map(|o| o.to_path_buf())
+                .unwrap_or_else(|| target_file.clone()),
+        };
```

Comment:
Can we pull all of this code out into its own `construct_manifest()` function? That way the `Init::execute()` function isn't filled with 100 lines of copying variables from `self` and `cargo_toml` into a `wapm_toml::Manifest`.